libevmasm/SemanticInformation.cpp

```diff
@@ -208,6 +208,27 @@ bool SemanticInformation::movable(Instruction _instruction)
 	return true;
 }
 
+bool SemanticInformation::sideEffectFree(Instruction _instruction)
+{
+	// These are not really functional.
+	assertThrow(!isDupInstruction(_instruction) && !isSwapInstruction(_instruction), AssemblyException, "");
+
+	InstructionInfo info = instructionInfo(_instruction);
+	switch (_instruction)
+	{
+	// All the instructions that merely read memory are fine
+	// even though they are marked "sideEffects" in Instructions.cpp
```

Comment:
*Why* are they marked "sideEffects" there :-)?

Reply:
It's just a different notion of side effect...

Reply:
Ah and to be more exact: The side effect is that `msize` changes, but we ignore that here.